src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinSource.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.conversion;
+
+import java.util.Map;
+
+/**
+ * Provider interface to obtain and store information from domain class.
+ * For Vertex and Edge, they consist of id (String, Reserved), label (String, Reserved) and
+ * a set of properties.
+ * The property key should be String, and value can be one of String, number and boolean.
+ */
+public interface GremlinSource {
+    /**
+     * Set the id (String) of domain
+     *
+     * @return never
+     */
+    void setId(String id);
+
+    /**
+     * Set the label (String) of domain
+     *
+     * @return never
+     */
+    void setLabel(String label);
+
+    /**
+     * Set the property map (String) of domain
```

Comment:
Not sure whether we need all these comments for these setters or not, as the setters name and signature is straight forward enough, adding comments may require future maintenance if method signature changed.